bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/channelhelper/CameraChannelHelper.java

```diff
@@ -47,53 +48,61 @@ public CameraChannelHelper(Set<String> providedGroups) {
     public void setUrls(String vpnUrl, @Nullable String localUrl) {
         this.localUrl = localUrl;
         this.vpnUrl = vpnUrl;
-        this.isLocal = localUrl != null;
-    }
-
-    public @Nullable String getLocalURL() {
-        return localUrl;
     }
 
     @Override
     protected @Nullable State internalGetProperty(String channelId, NAThing naThing, Configuration config) {
         if (naThing instanceof HomeStatusModule camera) {
-            boolean isMonitoring = OnOffType.ON.equals(camera.getMonitoring());
-            switch (channelId) {
-                case CHANNEL_MONITORING:
-                    return camera.getMonitoring();
-                case CHANNEL_SD_CARD:
-                    return toStringType(camera.getSdStatus());
-                case CHANNEL_ALIM_STATUS:
-                    return toStringType(camera.getAlimStatus());
-                case CHANNEL_LIVEPICTURE_VPN_URL:
-                    return toStringType(getLivePictureURL(false, isMonitoring));
-                case CHANNEL_LIVEPICTURE_LOCAL_URL:
-                    return toStringType(getLivePictureURL(true, isMonitoring));
-                case CHANNEL_LIVEPICTURE:
-                    return toRawType(getLivePictureURL(isLocal, isMonitoring));
-                case CHANNEL_LIVESTREAM_VPN_URL:
-                    return getLiveStreamURL(false, (String) config.get(QUALITY_CONF_ENTRY), isMonitoring);
-                case CHANNEL_LIVESTREAM_LOCAL_URL:
-                    return getLiveStreamURL(true, (String) config.get(QUALITY_CONF_ENTRY), isMonitoring);
-            }
+            return switch (channelId) {
+                case CHANNEL_MONITORING -> camera.getMonitoring();
+                case CHANNEL_SD_CARD -> toStringType(camera.getSdStatus());
+                case CHANNEL_ALIM_STATUS -> toStringType(camera.getAlimStatus());
+                default -> liveChannels(channelId, (String) config.get(QUALITY_CONF_ENTRY), camera,
+                        OnOffType.ON.equals(camera.getMonitoring()), localUrl != null);
+            };
         }
         return null;
     }
 
-    private @Nullable String getLivePictureURL(boolean local, boolean isMonitoring) {
-        String url = local ? localUrl : vpnUrl;
-        if (!isMonitoring || (local && !isLocal) || url == null) {
+    public @Nullable String getLivePictureURL(boolean local, boolean isMonitoring) {
+        if (!isMonitoring || (local && (localUrl == null))) {
             return null;
         }
-        return "%s%s".formatted(url, LIVE_PICTURE);
+        return "%s%s".formatted(getUrl(local), LIVE_PICTURE);
```

Comment:
Please keep a check that `getUrl(local)` is not null, something like:
```
String url = getUrl(local);
if (!isMonitoring || (url == null) {
    return null;
}
return "%s%s".formatted(url, LIVE_PICTURE);
```

Reply:
Done